Sources/SpyableMacro/Factories/FunctionImplementationFactory.swift

```diff
@@ -142,3 +142,11 @@ struct FunctionImplementationFactory {
         )
     }
 }
+
+private extension DeclModifierListSyntax {
+    var removingMutating: Self {
+        filter {
+            $0.name.text != "mutating"
```

Comment:
The code is perfectly valid but instead of writing the `text` string manually I would consider using a static function on `TokenSyntax`. What do you think of this approach?
```suggestion
            $0.name.text != TokenSyntax.keyword(.mutating).text
```

> [!IMPORTANT]
> At first glance, one might think it's logical to compare like this:
> ```swift
> $0.name != .keyword(.mutating)
> ```
> However, that would be a mistake. `TokenSyntax` includes leading and trailing trivia which play a crucial role in the comparison. I learned this the hard way.... :/

Reply:
Ah that's much nicer! I was honestly fussing about with some if case let syntax with the `TokenSyntax` that wasn't going my way before settling on what I committed. Your solution is perfect.
And good tip about `TokenSyntax`'s trivia, thanks!